contents/tutorials/vue-surveys.md

```diff
@@ -0,0 +1,623 @@
+        @onSubmit="handleSubmit"
+      />
+    </div>
+  </main>
+</template>
+
+<script>
+import CustomSurvey from './components/CustomSurvey.vue'
+
+export default {
+  name: 'App',
+  components: {
+    CustomSurvey
+  },
+  data() {
+    return {
+      showSurvey: true
+    };
+  },
+  methods: {
+    handleDismiss() {
+      this.showSurvey = false;
+    },
+    handleSubmit(value) {
+      console.log("Submitted value:", value);
+      this.showSurvey = false;
+    }
+  }
+}
+</script>
+```
+
+This shows a survey popup every time you visit your app's homepage.
+
+![Custom survey UI](../images/tutorials/nextjs-surveys/sample-survey-ui.png)
+
+#### 2. Fetch the survey from PostHog
+
+PostHog keeps track of all active surveys for a user (this is especially helpful if you set up [custom targeting options](/docs/surveys/creating-surveys#targeting)). 
+
+To fetch the active surveys, we use `this.$posthog.getActiveMatchingSurveys()`:
+
+```vue file=App.vue
+<template>
+  <!-- ... rest of your template ... -->
+</template>
+
+<script>
+import CustomSurvey from './components/CustomSurvey.vue'
+
+export default {
+  name: 'App',
+  components: {
+    CustomSurvey
+  },
+  data() {
+    return {
+      showSurvey: true,
+    };
+  },
+  mounted() {
+    this.$posthog.getActiveMatchingSurveys((surveys) => {
+    });
+  },
+  methods: {
+    // ... rest of your methods ...
+  }
+}
+</script>
+
+<style>
+  /* ... rest of your styles ... */
+</style>
+
+```
+
+`posthog.getActiveMatchingSurveys()` returns a surveys object that looks like this:
+
+```JSON
+[
+   {
+     "id": "018cfcd5-107e-0000-49a1-8e7c6b825947",
+     "name": "Net promoter score (NPS) API Survey",
+     "description": "",
+     "type": "api",
+     "linked_flag_key": null,
+     "targeting_flag_key": null,
+     "questions": [
+       {
+         "type": "rating",
+         "scale": 10,
+         "display": "number",
+         "question": "How likely are you to recommend us to a friend?",
+         "description": "",
+         "lowerBoundLabel": "Unlikely",
+         "upperBoundLabel": "Very likely"
+       }
+     ],
+     "conditions": null,
+     "start_date": "2024-01-12T08:41:20.614000Z",
+     "end_date": null
+   }
+]
+```
+
+We can use this survey object to configure our `CustomSurvey` component:
+
+```vue file=App.vue
+<template>
+  <main>
+    <div class="App">
+      <h1>This is our Vue.js survey tutorial</h1>
+      <CustomSurvey
+        v-if="showSurvey"
+        :title="surveyTitle"
+        @onDismiss="handleDismiss"
+        @onSubmit="handleSubmit"
+      />
+    </div>
+  </main>
+</template>
+
+<script>
+import CustomSurvey from './components/CustomSurvey.vue'
+
+export default {
+  name: 'App',
+  components: {
+    CustomSurvey
+  },
+  data() {
+    return {
+      showSurvey: true,
+      surveyTitle: '',
+      surveyID: ''
+    };
+  },
+  mounted() {
+    this.fetchActiveSurveys();
+  },
+  methods: {
+    fetchActiveSurveys() {
```

Comment:
Why not just keep this method in `mounted`? It is slightly inconsistent with last section.

Reply:
Will update previous section.